src/expression/function_expression.cpp

```diff
@@ -42,13 +42,17 @@ expression::FunctionExpression::FunctionExpression(
 
 type::Value FunctionExpression::Evaluate(
     const AbstractTuple *tuple1, const AbstractTuple *tuple2,
-    UNUSED_ATTRIBUTE executor::ExecutorContext *context) const {
+    executor::ExecutorContext *context) const {
   std::vector<type::Value> child_values;
 
   PELOTON_ASSERT(func_.impl != nullptr);
   for (auto &child : children_) {
     child_values.push_back(child->Evaluate(tuple1, tuple2, context));
   }
+  if (func_name_ == "nextval" || func_name_ == "currval") {
+    uint64_t ctx = (uint64_t)context;
```

Comment:
Casting a pointer to a 64bit integer and passing as an BigIntValue doesn't look like a good design for me.